src/main/java/com/limechain/trie/structure/TrieStructure.java

```diff
@@ -501,6 +506,37 @@ public boolean removeValueAtKey(Nibbles key) {
         };
     }
 
+    public int clearNodeValueRecursive(TrieNodeIndex nodeIndex, Long limit) {
+        return clearNodeValueRecursive(nodeIndex.getValue(), limit);
+    }
+
+    public int clearNodeValueRecursive(int nodeIndex, Long limit) {
```

Comment:
This shouldn't be public, the design premise being that the public interface only exposes access through `TrieNodeIndex`es and no raw `int`s.